cmd/fauxrpc/cmd_curl.go

```diff
@@ -0,0 +1,248 @@
+				AllowHTTP: true,
+				DialTLS: func(network, addr string, _ *tls.Config) (net.Conn, error) {
+					// If you're also using this client for non-h2c traffic, you may want
+					// to delegate to tls.Dial if the network isn't TCP or the addr isn't
+					// in an allowlist.
+					return net.Dial(network, addr)
+				},
+				// Don't forget timeouts!
+			},
+		}
+	}
+
+	// Determine the base URL
+	baseURL := c.Addr
+
+	// 3. Discover and filter methods
+	methodsToCall := make(map[string]protoreflect.MethodDescriptor)
+
+	if c.Method != "" {
+		// User provided a specific service or method
+		if strings.Contains(c.Method, "/") {
+			// Full method name provided
+			serviceName := c.Method[:strings.LastIndex(c.Method, "/")]
+			methodName := c.Method[strings.LastIndex(c.Method, "/")+1:]
+
+			serviceDesc := srv.Get(serviceName)
+			if serviceDesc == nil {
+				return fmt.Errorf("service %s not found", serviceName)
+			}
+			methodDesc := serviceDesc.Methods().ByName(protoreflect.Name(methodName))
+			if methodDesc == nil {
+				return fmt.Errorf("method %s not found in service %s", methodName, serviceName)
+			}
+			if !methodDesc.IsStreamingClient() && !methodDesc.IsStreamingServer() {
+				methodsToCall[c.Method] = methodDesc
+			} else {
+				slog.Warn("Skipping streaming method", "method", c.Method)
+			}
+		} else {
+			// Service name provided, call all non-streaming methods in that service
+			serviceName := c.Method
+			serviceDesc := srv.Get(serviceName)
+			if serviceDesc == nil {
+				return fmt.Errorf("service %s not found", serviceName)
+			}
+			for i := 0; i < serviceDesc.Methods().Len(); i++ {
+				methodDesc := serviceDesc.Methods().Get(i)
+				if !methodDesc.IsStreamingClient() && !methodDesc.IsStreamingServer() {
+					fullMethodName := fmt.Sprintf("%s/%s", serviceDesc.FullName(), methodDesc.Name())
+					methodsToCall[fullMethodName] = methodDesc
+				} else {
+					slog.Warn("Skipping streaming method", "method", methodDesc.FullName())
+				}
+			}
+		}
+	} else {
+		// Call all non-streaming methods in all discovered services
+		srv.ForEachService(func(serviceDesc protoreflect.ServiceDescriptor) bool {
+			for i := 0; i < serviceDesc.Methods().Len(); i++ {
+				methodDesc := serviceDesc.Methods().Get(i)
+				if !methodDesc.IsStreamingClient() && !methodDesc.IsStreamingServer() {
+					fullMethodName := fmt.Sprintf("%s/%s", serviceDesc.FullName(), methodDesc.Name())
+					methodsToCall[fullMethodName] = methodDesc
+				} else {
+					slog.Warn("Skipping streaming method", "method", methodDesc.FullName())
+				}
+			}
+			return true
+		})
+	}
+
+	if len(methodsToCall) == 0 {
+		return fmt.Errorf("no non-streaming methods found to call")
+	}
+
+	fakers := []fauxrpc.ProtoFaker{
+		stubs.NewStubFaker(srv.StubDatabase),
+		fauxrpc.NewFauxFaker(),
+	}
+	faker := fauxrpc.NewMultiFaker(fakers)
+
+	ctx := context.Background()
+
+	// 4. Make RPC calls
+	for fullMethodName, methodDesc := range methodsToCall {
```

Comment:
The inside of this block should be moved to a method. This is hinted at by the fact that there's so many continues